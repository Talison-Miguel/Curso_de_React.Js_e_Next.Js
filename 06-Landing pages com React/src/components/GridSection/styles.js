import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const Grid = styled.div`
    ${({ theme }) => css``}
`;

export const GridElement = styled.div`
    ${({ theme }) => css``}
`;
