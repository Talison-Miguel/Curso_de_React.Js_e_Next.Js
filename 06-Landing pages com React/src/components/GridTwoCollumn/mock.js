export default {
    title: 'Grid two colums',
    text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur corporis, explicabo aspernatur facere, itaque a, neque ab optio ducimus ipsa quasi aliquid tempora incidunt atque enim magni ex voluptate modi.`,
    srcImg: 'assets/images/javascript.svg',
};
