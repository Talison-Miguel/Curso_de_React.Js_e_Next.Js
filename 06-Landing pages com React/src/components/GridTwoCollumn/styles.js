import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
    ${({ theme }) => css``}
`;
