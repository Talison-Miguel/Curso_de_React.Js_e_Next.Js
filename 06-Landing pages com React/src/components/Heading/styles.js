import styled, { css } from 'styled-components';

const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.medium};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.large};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.xlarge};
    `,
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lessThanMedium} {
        font-size: ${theme.font.sizes.xlarge};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, colordark, size, uppercase }) => css`
        color: ${colordark ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)}
        ${titleCase(uppercase)}
    `}
`;
