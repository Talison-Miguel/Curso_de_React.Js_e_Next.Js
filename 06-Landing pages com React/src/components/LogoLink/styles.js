import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;
        display: flex;
        align-items: center;

        > img {
            max-height: 3rem;
        }
    `}
`;
