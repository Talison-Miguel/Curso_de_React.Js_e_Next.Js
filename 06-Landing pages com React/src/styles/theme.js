export const theme = {
    colors: {
        primaryColor: '#0a1128',
        secondaryBg: '#dc143c',
        white: '#fff',
    },
    font: {
        family: {
            default: "'Open Sans', sans-serif",
            secondary: "'Montserrat', sans-serif",
        },
        sizes: {
            xmall: '8rem',
            small: '1.6rem',
            medium: '2.4rem',
            large: '3.2rem',
            xlarge: '4.0rem',
            xxlarge: '4.8rem',
            huge: '5.6rem',
            xhuge: '6.4rem',
        },
    },
    media: {
        lessThanMedium: '(max-width: 768px)',
    },
    spacings: {
        xmall: '8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
    },
};
