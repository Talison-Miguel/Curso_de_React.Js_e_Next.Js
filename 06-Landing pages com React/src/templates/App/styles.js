import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    ${({ theme }) => css`
        background-color: ${theme.colors.secondaryBg};
    `}
`;
