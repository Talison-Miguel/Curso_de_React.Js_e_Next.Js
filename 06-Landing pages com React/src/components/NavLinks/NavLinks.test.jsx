import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { NavLinks } from '.';

import mock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
    it('Should render links', () => {
        renderTheme(<NavLinks links={mock} />);

        expect(screen.getAllByRole('link')).toHaveLength(mock.length);
    });

    it('Should not render links', () => {
        renderTheme(<NavLinks />);

        expect(screen.queryAllByText(/links/i)).toHaveLength(0);
    });

    it('Should render flex-flow row links', () => {
        renderTheme(<NavLinks links={mock} />);

        expect(screen.getByText(/link 10/i).parentElement).toHaveStyleRule('flex-flow', 'row wrap');
    });

    it('Should math snapshot', () => {
        const { container } = renderTheme(<NavLinks links={mock} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
