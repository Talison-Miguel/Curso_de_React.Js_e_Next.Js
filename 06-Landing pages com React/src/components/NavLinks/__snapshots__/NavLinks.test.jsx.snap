// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> Should math snapshot 1`] = `
.c0 {
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background: #dc143c;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;
